app/src/main/java/org/dhis2/usescases/qrScanner/ScanActivity.kt

```diff
@@ -0,0 +1,103 @@
+package org.dhis2.usescases.qrScanner
+
+import android.Manifest
+import android.app.Activity
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.os.Bundle
+import androidx.core.app.ActivityCompat
+import androidx.core.content.ContextCompat
+import androidx.databinding.DataBindingUtil
+import com.google.zxing.Result
+import me.dm7.barcodescanner.zxing.ZXingScannerView
+import org.dhis2.App
+import org.dhis2.R
+import org.dhis2.databinding.ActivityQrBinding
+import org.dhis2.usescases.general.ActivityGlobalAbstract
+import org.dhis2.utils.Constants
+import javax.inject.Inject
+
+class ScanActivity : ActivityGlobalAbstract(), ZXingScannerView.ResultHandler {
+    private lateinit var binding: ActivityQrBinding
+    private lateinit var mScannerView: ZXingScannerView
+    private var isPermissionRequested = false
+    private var optionSetUid: String? = null
+
+    @Inject
+    lateinit var scanPresenter: ScanPresenter
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        optionSetUid = intent.getStringExtra(Constants.OPTION_SET)
+        (applicationContext as App)
+            .userComponent()
+            ?.plus(ScanModule(optionSetUid))
+            ?.inject(this)
+
+        binding = DataBindingUtil.setContentView(this, R.layout.activity_qr)
```

Comment:
Maybe it is a good idea to also change the name of the layout to correspond to the activity's name? What do you think?

Reply:
You are right